### ISSUE TITLE
`ipfs key gen` expect mandatorily the optional parameter '-t'

### ISSUE THREAD

**Issue by RubenKelevra** [ipfs/go-ipfs#6861]
#### Version information:
go-ipfs version: 0.4.23-6ce9a355f
Repo version: 7
System version: amd64/linux
Golang version: go1.13.7

#### Description:

The help states, that '-t' is optional:

```
$ ipfs key gen --help
USAGE
  ipfs key gen <name> - Create a new keypair

SYNOPSIS
  ipfs key gen [--type=<type> | -t] [--size=<size> | -s] [--] <name>

ARGUMENTS

  <name> - name of key to create

OPTIONS

  -t, --type  string - type of the key to create [rsa, ed25519].
  -s, --size  int    - size of the key to generate.
```

But running key gen without '-t' fails:

```
$ ipfs key gen "test"
Error: please specify a key type with --type
```

**Comment by Stebalien** [comment]
In general, IPFS doesn't have a way to specify "optional" versus "mandatory" flags. However, it makes sense to just make all flags optional (and that's much easier to fix than adding mandatory flags).

fix in #6864.

**Comment by RubenKelevra** [comment]
> In general, IPFS doesn't have a way to specify "optional" versus "mandatory" flags. However, it makes sense to just make all flags optional (and that's much easier to fix than adding mandatory flags).

While the `--help` output isn't strictly speaking a man page, its output follows the styling of a man page.

The man page for `man` specifies that 

> any or all arguments within [ ] are optional.

That was what I was referring to.

> fix in #6864.

Thanks for the fast fix!

**Comment by Stebalien** [comment]
> That was what I was referring to.

Yeah, I know. But that help text is auto-generated and fixing it would require either a hack or adding the concept of "mandatory"/"optional" flags to our commands library (non trivial and really not worth it).
